openlibrary/catalog/marc/parse.py

```diff
@@ -292,7 +294,24 @@ def read_languages(rec):
         return
     found = []
     for f in fields:
-        found += [i.lower() for i in f.get_subfield_values('a') if i and len(i) == 3]
+        is_translation = (
+            f.ind1() == '1'
+        )  # It can be a translation even without the original language being coded
+        if f.ind2() == '7':
+            code_source = ' '.join(f.get_subfield_values('2'))
+            # TODO: What's the best way to handle these?
+            raise MarcException("Non-MARC language code(s), source = ", code_source)
+            continue  # Skip anything which is using a non-MARC code source e.g. iso639-1
+        for i in f.get_subfield_values('a'):
+            if (
+                i
+            ):  # This is carried over from previous code, but won't it always be true?
```

Comment:
I _think_ this was trying to catch an empty string, which is a possibility in `008` , but probably not here. Relates to my comment about wanting to unify language code extraction from both `008` and `041`  so we are processing the codes consistently and in one place.

Reply:
This test is actually redundant with the length check below, so I've removed it.